Source/WebCore/html/HTMLDetailsElement.cpp

```diff
@@ -117,6 +113,14 @@ void HTMLDetailsElement::didAddUserAgentShadowRoot(ShadowRoot& root)
 
     m_defaultSlot = HTMLSlotElement::create(slotTag, document());
     ASSERT(!hasAttribute(openAttr));
+    m_defaultSlot->setInlineStyleProperty(CSSPropertyContentVisibility, CSSValueHidden);
+    m_defaultSlot->setInlineStyleProperty(CSSPropertyDisplay, CSSValueBlock);
+    root.appendChild(*m_defaultSlot);
+
+    static MainThreadNeverDestroyed<const String> stylesheet(StringImpl::createWithoutCopying(detailsElementShadowUserAgentStyleSheet));
```

Comment:
Maybe we can use `adoptedStyleSheets()` instead of creating a new style element?

Reply:
The spec uses a style element: https://html.spec.whatwg.org/#the-details-and-summary-elements

Reply:
It should be the same though. I said as much in the specification review and the main reason the `style` element approach was allowed because you couldn't observe the difference. If you're setting it up programmatically I'm not entirely sure it makes sense to create the additional node.

Reply:
I can try it, but it'll probably end up inconsistent with meter/attachment, although I guess they were created before `adoptedStylesheets`

Reply:
I tried to address this but none of my attempts worked